internal/servers/destinations.go

```diff
@@ -76,12 +76,26 @@ func (s *DestinationServer) Write2(msg pb.Destination_Write2Server) error {
 	if err := json.Unmarshal(r.Tables, &tables); err != nil {
 		return status.Errorf(codes.InvalidArgument, "failed to unmarshal tables: %v", err)
 	}
+	var sourceSpec specs.Source
+	if r.SourceSpec == nil {
+		// this is for backward compatibility
+		sourceSpec = specs.Source{
+			Name: r.Source,
+		}
+	} else {
+		if err := json.Unmarshal(r.SourceSpec, &sourceSpec); err != nil {
+			return status.Errorf(codes.InvalidArgument, "failed to unmarshal source spec: %v", err)
+		}
+	}
 	sourceName := r.Source
 	syncTime := r.Timestamp.AsTime()
+	sourceSpec = specs.Source{
+		Name: sourceName,
+	}
```

Comment:
Why are we overwriting `sourceSpec`, shortly after unmarshaling from `r.SourceSpec`?

Reply:
good catch. a copy paste from the previous if